admiral/pkg/clusters/serviceentry.go

```diff
@@ -192,18 +200,20 @@ func modifyServiceEntryForNewServiceOrPod(
 		return nil, nil
 	}
 
+	// This is done for optimizing dependency handler. If a new asset onboards to a service, we want to create/update the SE only in that dependent's cluster.
+	// START dependency handler optimization
 	var createResourcesOnlyInDependentOverrideClusters bool
 	dependentClusterOverride, ok := ctx.Value(common.DependentClusterOverride).(*common.Map)
 	if !ok {
 		ctxLogger.Warnf(common.CtxLogFormat, "event", "", "", "", "dependent cluster override not passed")
 	} else {
-		if dependentClusterOverride != nil && len(dependentClusterOverride.GetKeys()) > 0 {
+		if dependentClusterOverride != nil && len(dependentClusterOverride.GetValues()) > 0 {
```

Comment:
why this change made?

Reply:
Its because [this method](https://github.com/istio-ecosystem/admiral/blob/f7338904d9325d0576fde641cae1e72966191fef/admiral/pkg/controller/common/types.go#L422-L430) returns the values instead of the keys. I think this was a bug but it continued to work as the Cache entries  have the same key and value. In my case the cache entry looks like this - 

```
asset -> env -> workloadEnv -> warmupDuration
```
When I called getKeys() I expected an array of workloadEnvs, but I was getting the list of warmupDurations.